File: events/get_engagement_metrics.js
// this sample JavaScript code retrieves email engagement metrics that can be used in data analysis.
// the first example gets all Opened Email events.
// the second example gets all Clicked Email events for a specific timeframe.

import { ApiKeySession, EventsApi } from 'klaviyo-api'

// load private API key as an environment variable. Note that this API key needs to have read access to events.
// if using a .env file, ensure to load it using dotenv or similar package. 
const apiKey = process.env.KLAVIYO_API_KEY
if (!apiKey) {
  throw new Error('Please set the KLAVIYO_API_KEY environment variable.')
}

const session = new ApiKeySession(apiKey)
const eventsApi = new EventsApi(session)

// example 1: Get all Opened Email events
const openedEmailMetricId = 'Xj5uUn'

const openedEmailFilter = `equals(metric_id,'${openedEmailMetricId}')`
const openedEmailEvents = await eventsApi.getEvents({openedEmailFilter})
console.log('Opened Email Events:', openedEmailEvents)

// example 2: Get all Clicked Email events for a specific timeframe
const clickedEmailMetricId = 'UmeLw3'
const startDate = '2023-01-01T00:00:00Z'
const endDate = '2023-01-31T23:59:59Z'
const clickedEmailFilter = `and(equals(metric_id,'${clickedEmailMetricId}'),greater-or-equal(datetime,${startDate}),less-than(datetime,${endDate}))`

const clickedEmailEvents = await eventsApi.getEvents({filter: clickedEmailFilter})
console.log('Clicked Email Events:', clickedEmailEvents)

File: events/get_engagement_metrics.py
# this sample Python code retrieves email engagement metrics that can be used in data analysis.
# the first example gets all Opened Email events.
# the second example gets all Clicked Email events for a specific timeframe.
import os
from klaviyo_api import KlaviyoAPI
from datetime import datetime

# load your private API key from environment variables. 
# ensure you have set the environment variable "private_api_key" with Klaviyo private API key that has read access to events. 
private_api_key = os.environ.get("private_api_key")

# instantiate the KlaviyoAPI client with your private API key.
klaviyo = KlaviyoAPI(private_api_key)

# example 1: get all Opened Email events. Replace opened_email_metric_id with the Opened Email metric ID from your account.
opened_email_metric_id = 'Xj5uUn'

klaviyo.Events.get_events(filter=f"equals(metric_id,'{opened_email_metric_id}')")

# example 2: get all Clicked Email events from November 2024. Replace clicked_email_metric_id with the Clicked Email metric ID from your Klaviyo account. Replace the numbers in start_date and end_date with your dates of interest.
clicked_email_metric_id = 'UmeLw3'
start_date = datetime(2024,11,1,0,0,0)
end_date = datetime(2024,12,1,0,0,0)
start_date_str = start_date.strftime('%Y-%m-%dT%H:%M:%S')
end_date_str = end_date.strftime('%Y-%m-%dT%H:%M:%S')

klaviyo.Events.get_events(filter=f"and(equals(metric_id,'{clicked_email_metric_id}'),greater-or-equal(datetime,{start_date_str}),less-than(datetime,{end_date_str}))")